### PR TITLE
i2c_ll_stm32: Use macros to add I2C instances

### PR DESCRIPTION
**Pull request by ydamigos**
Use macros to add I2C instances

Signed-off-by: Yannis Damigos <giannis.damigos@gmail.com>

### CODE REVIEW

File: drivers/i2c/i2c_ll_stm32.c [hunk 1]
```diff
@@ -245,96 +245,78 @@ static int i2c_stm32_init(struct device *dev)
...
+static struct i2c_stm32_data i2c_stm32_dev_data_##name;			\
+									\
+DEVICE_AND_API_INIT(i2c_stm32_##name, CONFIG_##name##_NAME,		\
+		    &i2c_stm32_init, &i2c_stm32_dev_data_##name,	\
+		    &i2c_stm32_cfg_##name,				\
+		    POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE,	\
+		    &api_funcs);					\
+									\
+STM32_I2C_IRQ_HANDLER(name)
+
```
Comment (erwango) [inline drivers/i2c/i2c_ll_stm32.c]: To enhance readability, what about adding a comment here, to clearly separate macro declaration and instances declaration part?

File: drivers/i2c/i2c_ll_stm32.c [hunk 2]
```diff
@@ -245,96 +245,78 @@ static int i2c_stm32_init(struct device *dev)
...
+			    DEVICE_GET(i2c_stm32_##name), 0);		\
+		irq_enable(DT_##name##_EVENT_IRQ);			\
+									\
+		IRQ_CONNECT(DT_##name##_ERROR_IRQ,			\
+			    DT_##name##_ERROR_IRQ_PRI,			\
+			    stm32_i2c_error_isr,			\
+			    DEVICE_GET(i2c_stm32_##name), 0);		\
+		irq_enable(DT_##name##_ERROR_IRQ);			\
+	} while (0)
 #endif
```
Comment (erwango) [inline drivers/i2c/i2c_ll_stm32.c]: /* CONFIG_I2C_STM32_COMBINED_INTERRUPT */

File: drivers/i2c/i2c_ll_stm32.c [hunk 3]
```diff
@@ -245,96 +245,78 @@ static int i2c_stm32_init(struct device *dev)
...
+#define STM32_I2C_IRQ_HANDLER(name)					\
+static void i2c_stm32_irq_config_func_##name(struct device *dev)	\
+{									\
+	STM32_I2C_IRQ_CONNECT_AND_ENABLE(name);				\
 }
+#else
+#define STM32_I2C_IRQ_HANDLER_DECL(name)
+#define STM32_I2C_IRQ_HANDLER_FUNCTION(name)
+#define STM32_I2C_IRQ_HANDLER(name)
 #endif
```
Comment (erwango) [inline drivers/i2c/i2c_ll_stm32.c]: /* CONFIG_I2C_STM32_INTERRUPT */

File: drivers/i2c/i2c_ll_stm32.c [hunk 4]
```diff
@@ -245,96 +245,78 @@ static int i2c_stm32_init(struct device *dev)
...
+#define STM32_I2C_IRQ_HANDLER_DECL(name)				\
+static void i2c_stm32_irq_config_func_##name(struct device *dev)
+#define STM32_I2C_IRQ_HANDLER_FUNCTION(name)				\
+	.irq_config_func = i2c_stm32_irq_config_func_##name,
+#define STM32_I2C_IRQ_HANDLER(name)					\
+static void i2c_stm32_irq_config_func_##name(struct device *dev)	\
+{									\
+	STM32_I2C_IRQ_CONNECT_AND_ENABLE(name);				\
 }
+#else
```
Comment (erwango) [inline drivers/i2c/i2c_ll_stm32.c]: Break a line here

File: drivers/i2c/i2c_ll_stm32.c [hunk 5]
```diff
@@ -245,96 +245,78 @@ static int i2c_stm32_init(struct device *dev)
...
+#define STM32_I2C_IRQ_HANDLER_DECL(name)
+#define STM32_I2C_IRQ_HANDLER_FUNCTION(name)
+#define STM32_I2C_IRQ_HANDLER(name)
```
Comment (erwango) [inline drivers/i2c/i2c_ll_stm32.c]: Break a line here

**Review by erwango — CHANGES_REQUESTED** [review]
Nitpicks to enhance readability

**Comment by ydamigos** [comment]
@erwango Thanks for the review. Comments addressed.

**Review by erwango — APPROVED** [review]

**Pull request merged**